tests/SprykerSdkTest/Spryk/Console/SprykRunSpryksTest.php

```diff
@@ -46,4 +46,46 @@ public function testExecutesSpryks(): void
         $this->assertDirectoryExists($this->tester->getVirtualDirectory() . 'vendor/spryker/spryker/Bundles/FooBar/firstDirectory');
         $this->assertDirectoryExists($this->tester->getVirtualDirectory() . 'vendor/spryker/spryker/Bundles/FooBar/secondDirectory');
     }
+
+    /**
+     * @return void
+     */
+    public function testExecutesNotUniqueSpryks(): void
```

Comment:
The method name is not clear to me and doesn't explain what this method tests.

You can also add a doc block on top to make the role of this method clear.

Reply:
This test was added for code coverage. It checks how spryk wrapper works if duplicate spryks are specified in the spryks attribute. In this case, only one spryk is executed, but I can't really check it, because running it twice doesn't throw an error and doesn't create artifacts. I tried to use a SprykExecutor mock to count the number of buildSpryk calls, but SprykExecutor is a private service and cannot be replaced during tests. Right now it looks like this test only checks that spryk has at least worked. Should I remove it or can I leave it as is?

Reply:
You can always outline in the DocBlock of the method what it exactly covers when the method name would become extremely long.

You can also mock private Service in the Symfony Container. For this you can use `\SprykerSdkTest\Module\SprykHelper::setDependency()`.

But there should be nothing to ensure that a Spryk can only be executed once only when they are exactly the same with all arguments. But usually, you will add the same Spryk multiple times but with a different configuration.

I guess from what you wrote that a good method name would be `testExecuteOnlyRunsUniqueSpryks`. A DocBlock description could then be:

```
Tests that when a Spryk with the same arguments was already executed it will not be executed again. It might happen in the whole chain of Spryks with pre and postSpryks that one defines exactly the same Spryk. Running it only once is ensured because of performance reasons.
```

Reply:
Renamed method, added phpdoc, but I still can't mock SprykExecutor. 
```
$executorMock = $this->createPartialMock(SprykExecutor::class, ['buildSpryk']);
$executorMock->expects(static::once())->method('buildSpryk');

$this->tester->setDependency('SprykerSdk\\Spryk\\Model\\Spryk\\Executor\\SprykExecutor', $executorMock);
```
returns error:

"The "SprykerSdk\Spryk\Model\Spryk\Executor\SprykExecutor" service is private, you cannot replace it.". 

I will continue to investigate it.

Reply:
I double-checked and found that for this case we need to make the service public in services.php as it was done with others. Maybe you can find a better approach e.g. making services only public in tests?